File: sdk/ocaml/lib/vcloudstack.ml
(** vCloudStack Public Cloud: rescue, unrescue, console, usage *)

(** POST /api/service/{id}/vms/{vmid}/rescue - Reboot VM into rescue mode *)
let rescue_vm client params =
  HttpClient.post client "/api/service/{id}/vms/{vmid}/rescue" params

(** POST /api/service/{id}/vms/{vmid}/unrescue - Disable rescue mode *)
let unrescue_vm client params =
  HttpClient.post client "/api/service/{id}/vms/{vmid}/unrescue" params

(** GET /api/service/{id}/vms/{vmid}/console - Get VM console *)
let get_console client params =
  HttpClient.get client "/api/service/{id}/vms/{vmid}/console" params

(** GET /api/service/{id}/vms/{vmid}/usage - Get VM usage details *)
let get_usage client params =
  HttpClient.get client "/api/service/{id}/vms/{vmid}/usage" params

File: sdk/julia/src/vcloudstack.jl
# vCloudStack Public Cloud: rescue, unrescue, console, usage
struct VCloudStackResource
    http::HttpClient
end

# POST /api/service/{id}/vms/{vmid}/rescue - Reboot VM into rescue mode
function rescue_vm(r::VCloudStackResource; kwargs...)
    params = Dict(kwargs)
    http_post(r.http, "/api/service/{id}/vms/{vmid}/rescue"; params = params)
end

# POST /api/service/{id}/vms/{vmid}/unrescue - Disable rescue mode
function unrescue_vm(r::VCloudStackResource; kwargs...)
    params = Dict(kwargs)
    http_post(r.http, "/api/service/{id}/vms/{vmid}/unrescue"; params = params)
end

# GET /api/service/{id}/vms/{vmid}/console - Get VM console
function get_console(r::VCloudStackResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/service/{id}/vms/{vmid}/console"; params = params)
end

# GET /api/service/{id}/vms/{vmid}/usage - Get VM usage details
function get_usage(r::VCloudStackResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/service/{id}/vms/{vmid}/usage"; params = params)
end
